alf/algorithms/distributed_off_policy_algorithm.py

```diff
@@ -241,21 +239,26 @@ def receive_experience_data(replay_buffer: ReplayBuffer,
                 addr = 'tcp://' + ':'.join([unroller_ip, str(unroller_port)])
                 # Connect to an additional ROUTER
                 socket.connect(addr)
+        except queue.Empty:
+            pass
+
         if socket is not None:
             # Receive data from any router
             unroller_id, message = socket.recv_multipart()
-            if message == UnrollerMessage.EXP_SEG_END.encode():
+
+            buffer = io.BytesIO(message)
+            exp_params = torch.load(buffer, map_location='cpu')
+            # Use a temp buffer to store the received exps
+            if unroller_id not in unroller_exps_buffer:
+                unroller_exps_buffer[unroller_id] = []
+            unroller_exps_buffer[unroller_id].append(exp_params)
+
+            if int(exp_params.step_type) == StepType.LAST:
                 # Add the temp exp buffer to the replay buffer
-                for exp_params in unroller_exps_buffer[unroller_id]:
+                for i, exp_params in enumerate(
+                        unroller_exps_buffer[unroller_id]):
```

Comment:
If the batch size of the replay buffer is 1. env_id has to be 0 at the next line

Reply:
> If the batch size of the replay buffer is 1. env_id has to be 0 at the next line

This is true for the current assumption. But since exp_params always contains env_id, we can just use it. Do you mean we should assert it's equal to 0?

Reply:
We can just set it to 0 here?

Reply:
> We can just set it to 0 here?

updated